frontend/src/pages/PipelineDetails.tsx

```diff
@@ -288,14 +300,26 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
                     )}
 
                     <Graph
-                      graph={this.state.graph}
+                      graph={_graph}
                       selectedNodeId={selectedNodeId}
                       onClick={id => this.setStateSafe({ selectedNodeId: id })}
                       onError={(message, additionalInfo) =>
                         this.props.updateBanner({ message, additionalInfo, mode: 'error' })
                       }
                     />
 
+                    <div className={commonCss.transitiveReductionSwitch}>
+                      <SwitchWithLabel
```

Comment:
I think it'll be better if we have a tooltip on the switch/ or next to the switch that explains how we are simplifying the graph. A short paragraph, probably with a link to more details is great.

Because, the simplified dependencies might mean passing parameters/artifacts between steps and without explaining how we are simplifying the graph, it's will be confusing why those dependencies do not show up.

Reply:
For tooltip, you may use https://github.com/kubeflow/pipelines/blob/768317aee33d1c00a46637549fc2753aa357e0ab/frontend/src/atoms/HelpButton.tsx#L23 or https://github.com/kubeflow/pipelines/blob/768317aee33d1c00a46637549fc2753aa357e0ab/frontend/src/components/CustomTable.tsx#L347

Reply:
Added a tooltip as part of the refactoring into a component 6d4c769c

Reply:
I used the same tooltip as in `HelpButton`, so this code pattern is now repeated in two places. I tried to implement a new atom `CardTooltip` but I had some issues with passing `props.children`. This is what I tried to implement:

```jsx
const NostyleTooltip = withStyles({
  tooltip: {
    backgroundColor: 'transparent',
    border: '0 none',
    color: color.secondaryText,
    fontSize: fontsize.base,
    maxWidth: 220,
  },
})(Tooltip);

interface CardTooltipProps {
  helpText?: ReactNode;
}
export const CardTooltip: React.FC<CardTooltipProps> = props => {
  return (
    <NostyleTooltip
      title={
        <Card>
          <CardContent>{props.helpText}</CardContent>
        </Card>
      }
      interactive={true}
      leaveDelay={400}
      placement='top'
    >
      {props.children}
    </NostyleTooltip>
  );
};
```

but I get an error in `{props.children}`. Apparently the `NostyleTooltip` cannot accept a `ReactNode`: `Type 'ReactNode' is not assignable to type 'ReactElement<any, string | ((props: any) => .......`. This is a bit beyond me, do you have any tip on how to fix it or debug further?

Reply:
@Bobgy I gave this a second look and I figured out what was wrong, see my latest commit: https://github.com/kubeflow/pipelines/pull/4925/commits/80f3f1dcdc33631f8476ffec289671e218905e17